SDDev.Net.GenericRepository/CosmosDB/GenericRepository.cs

```diff
@@ -239,6 +237,26 @@ public async override Task<ISearchResult<TModel>> Get(string query, ISearchModel
             return response;
         }
 
+        public async Task<int> Count(Expression<Func<TModel, bool>> predicate, string partitionKey = null)
+        {
+            var queryOptions = new QueryRequestOptions();
+
+            if (!string.IsNullOrEmpty(partitionKey))
+                queryOptions.PartitionKey = new PartitionKey(partitionKey);
+            else
+                Log.LogWarning($"Enabling Cross-Partition Query in repo {this.GetType().Name}");
+
+            var query = Client
+                .GetItemLinqQueryable<TModel>(requestOptions: queryOptions)
+                .Where(x => x.ItemType.Contains(typeof(TModel).Name)) //force filtering by Item Type
+                .Where(predicate)
+                .Select(x => x.Id);
+
+            var count = await query.CountAsync();

```

Comment:
Why await this and return instead of just making the method not async and returning the task?